internal/staging/debouncer.go

```diff
@@ -0,0 +1,45 @@
+package staging
+
+import (
+	"context"
+	"time"
+)
+
+type Debouncer struct {
+	timeChan chan time.Time
+}
+
+func NewDebouncer(ctx context.Context, timeout time.Duration, callback func()) *Debouncer {
+	d := &Debouncer{
+		timeChan: make(chan time.Time, 100),
+	}
+
+	go func() {
```

Comment:
nit: I think I'd prefer something like `Debouncer.Start()` to kick off this routine explicitly, but no big deal right now. I'd prefer to merge as is and change it later if we need to.